sdk/cosmos/azure-spring-data-cosmos-test/src/test/java/com/azure/spring/data/cosmos/ContainerLock.java

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.data.cosmos;
+
+import com.azure.cosmos.implementation.guava25.base.Stopwatch;
+import com.azure.spring.data.cosmos.core.CosmosTemplate;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.annotation.Version;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+
+public class ContainerLock {
+
+    private CosmosTemplate template;
```

Comment:
This works for `CosmosTemplate` - do you plan to include `ReactiveCosmosTemplate` in the same ContainerLock class, or would you prefer for a `ReactiveContainerLock` class as we update more tests ?

Reply:
CosmosTempate is always available, even in the reactive tests (and many reactive tests already inject CosmosTemplate), so I just standardized on that to keep this class simple.  I can add support for ReactiveCosmosTemplate as well as there are a few reactive tests that do not already pull in CosmosTemplate.

Reply:
Sounds good, so what it seems like is this will be case to case basis. If we need reactive cosmos template, we will pull it in.